### PR TITLE
Update go libs

### PR DESCRIPTION
**Pull request by kvaps**
required for https://github.com/clastix/cluster-api-control-plane-provider-kamaji/pull/78

```
go 1.20 --> 1.22
kamaji v0.4.0 --> v0.5.0
cluster-api v1.4.0-beta.2.0 --> v1.6.4
kubernetes v0.29.0 --> v0.30.0
controller-gen v0.12.0 --> v0.14.0
```

### CODE REVIEW

**Comment by prometherion** [comment]
I think we need to update also the Go version of the underlying GH actions.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -27,16 +27,16 @@ jobs:
       - name: Run golangci-lint
         uses: golangci/golangci-lint-action@v3.0.0
         with:
-          version: v1.53.2
+          version: v1.54.2
```
Comment (prometherion) [inline .github/workflows/ci.yaml:30]: May I ask you to update also the `golangci-lint` version in the Makefile?

**Review by prometherion — CHANGES_REQUESTED** [review]

**Comment by prometherion** [comment]
I know the problem with `controller-gen` and Go 1.22, we just need to bump it to v0.14.0

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (kvaps) [reply .github/workflows/ci.yaml:30]: sure, done

**Comment by prometherion** [comment]
This change is a bit intrusive 😄 

Please, let's generate manifests with `make release` and let's remove the `gocognit` ignore, you can double check with `make lint`.

**Comment by kvaps** [comment]
I had old controller-gen at my PC, now diff should be fixed

**Comment by kvaps** [comment]
aha, you're right

**Review by prometherion — APPROVED** [review]

**Pull request merged**